SQL/New_patches/2018-05-28_publication_schema.sql

```diff
@@ -0,0 +1,151 @@
+-- Publication Status
+CREATE TABLE `publication_status` (
+  `PublicationStatusID` int(2) unsigned NOT NULL AUTO_INCREMENT,
+  `Label` varchar(255) NOT NULL,
+  CONSTRAINT `PK_publication_status` PRIMARY KEY(`PublicationStatusID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+INSERT INTO publication_status (`Label`) VALUES ('Pending');
+INSERT INTO publication_status (`Label`) VALUES ('Approved');
+INSERT INTO publication_status (`Label`) VALUES ('Rejected');
+
+-- Main table
+CREATE TABLE `publication` (
+    `PublicationID` int(10) unsigned NOT NULL AUTO_INCREMENT,
+    `PublicationStatusID` int(2) unsigned NOT NULL default 1,
+    `LeadInvestigatorID` int(10) unsigned NOT NULL,
+    `UserID` int(10) unsigned NOT NULL,
+    `RatedBy` int(10) unsigned,
+    `DateProposed` date NOT NULL,
+    `DateRated` date default NULL,
+    `Title` varchar(255) NOT NULL,
+    `RejectedReason` varchar(255) default NULL,
+    `Description` text NOT NULL,
+    CONSTRAINT `PK_publication` PRIMARY KEY(`PublicationID`),
+    CONSTRAINT `FK_publication_UserID` FOREIGN KEY(`UserID`) REFERENCES `users` (`ID`),
+    CONSTRAINT `FK_publication_RatedBy` FOREIGN KEY(`RatedBy`) REFERENCES `users` (`ID`),
+    CONSTRAINT `FK_publication_PublicationStatusID` FOREIGN KEY(`PublicationStatusID`) REFERENCES `publication_status` (`PublicationStatusID`),
+    CONSTRAINT `FK_publication_LeadInvestigatorID` FOREIGN KEY(`LeadInvestigatorID`) REFERENCES `publication_collaborator` (`PublicationCollaboratorID`),
```

Comment:
table `publication_collaborator` needs to be created first before it is referenced

Reply:
solved